android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/runtime/HideSystemUIDisposableEffect.kt

```diff
@@ -0,0 +1,67 @@
+package com.stadiamaps.ferrostar.maplibreui.runtime
+
+import android.app.Activity
+import android.graphics.Color
+import android.view.Window
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.ui.graphics.toArgb
+import androidx.compose.ui.platform.LocalContext
+import androidx.core.view.WindowCompat
+import androidx.core.view.WindowInsetsCompat
+import androidx.core.view.WindowInsetsControllerCompat
+
+/** Get the Window for the current scene (Activity). */
+@Composable
+fun window(): Window? {
+  val context = LocalContext.current
+  return (context as? Activity)?.window ?: return null
+}
+
+/** Get the WindowInsetsController for the provided window. */
+@Composable
+fun windowInsetsController(window: Window): WindowInsetsControllerCompat {
+  return WindowCompat.getInsetsController(window, window.decorView)
+}
+
+/**
+ * A Composable effect that hides the system UI (status bar and navigation bar) when the Composable
+ * is first composed and restores the system UI when the Composable is disposed.
+ */
+@Composable
+fun SystemUIDisposableEffect() {
```

Comment:
Clever ;) I think the file name has a bit more descriptive type name though. Did you mean to name it `HideSystemUIDisposableEffect`?